File: src/main/java/dev/kensa/sentence/Acronym.java
package dev.kensa.sentence;

import java.util.Objects;

public class Acronym {

    public static Acronym of(String acronym, String meaning) {
        if (acronym == null || acronym.length() < 2) {
            throw new IllegalArgumentException(String.format("Acronyms must be at least 2 characters. [%s]", acronym));
        }

        return new Acronym(acronym, meaning);
    }

    private final String acronym;
    private final String meaning;

    private Acronym(String acronym, String meaning) {
        this.acronym = acronym;
        this.meaning = meaning;
    }

    public String acronym() {
        return acronym;
    }

    public String meaning() {
        return meaning;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Acronym)) {
            return false;
        }
        Acronym acronym1 = (Acronym) o;
        return acronym.equals(acronym1.acronym);
    }

    @Override
    public int hashCode() {
        return Objects.hash(acronym);
    }
}

File: src/main/java/dev/kensa/sentence/Acronym.kt
package dev.kensa.sentence

import java.util.*

class Acronym(val acronym: String, val meaning: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is Acronym) {
            return false
        }
        return acronym == other.acronym
    }

    override fun hashCode(): Int = Objects.hash(acronym)

    companion object {
        private const val maxLength = 2
        @JvmStatic
        fun of(acronym: String, meaning: String): Acronym {
            require((acronym.length >= maxLength)) { "Acronyms must be at least $maxLength characters. [$acronym]" }
            return Acronym(acronym, meaning)
        }
    }
}